### ISSUE TITLE
Shut down the server when rebuilding

### ISSUE THREAD

**Issue by getkey** [thgh/rollup-plugin-serve#16]
It would be nice if, when the Rollup is started in watch mode, the server would stop when the bundle is being rebuilt. That way if you refresh your browser before it's done, you won't be loading the outdated bundle.

It looks like [`ongenerate`](https://github.com/rollup/rollup/wiki/Plugins#creating-plugins) could have been used to be notified when the bundle is being rebuilt, but from my tests there is some delay between the file being changed and `ongenerate` firing. I am not sure if this delay is due to the the detection being slow or something else.

**Comment by sormy** [comment]
Proposed fix:

```patch
diff -ur rollup-plugin-serve.old/dist/index.cjs.js rollup-plugin-serve/dist/index.cjs.js
--- rollup-plugin-serve.old/dist/index.cjs.js	2018-12-12 15:38:12.000000000 -0500
+++ rollup-plugin-serve/dist/index.cjs.js	2018-12-12 15:39:52.000000000 -0500
@@ -9,6 +9,8 @@
 var mime = _interopDefault(require('mime'));
 var opener = _interopDefault(require('opener'));
 
+var server;
+
 function serve (options) {
   if ( options === void 0 ) options = { contentBase: '' };
 
@@ -70,7 +72,10 @@
   };
 
   // If HTTPS options are available, create an HTTPS server
-  var server;
+  if (server) {
+    // release previous server instance if rollup is reloading configuration on watch
+    server.close()
+  }
   if (options.https) {
     server = https.createServer(options.https, requestListener).listen(options.port);
   } else {
```

Let me know if you are ok with this change, I will submit PR.

**Comment by thgh** [comment]
I didn't know how to fix this, but #27 has an elegant solution 👍
